contracts/protocol/facets/ExchangeHandlerFacet.sol

```diff
@@ -804,6 +804,40 @@ contract ExchangeHandlerFacet is IBosonExchangeHandler, BuyerBase, DisputeBase {
                     twinReceipt.amount = twin.amount;
                     twinReceipt.tokenType = twin.tokenType;
 
+                    if (twin.tokenType == TokenType.NonFungibleToken) {
+                        // Get all ranges of twins that belong to the seller and to the same token address of the new twin to validate if range is available
+                        TokenRange[] storage twinRanges = lookups.twinRangesBySeller[seller.id][twin.tokenAddress];
+
+                        uint256 lastIndex = twinRanges.length - 1;
+
+                        // Find the range that contains the twin
+                        for (uint256 j = 0; j <= lastIndex; j++) {
+                            if (twinRanges[j].start <= tokenId && twinRanges[j].end >= tokenId) {
```

Comment:
> A minor optimization is possible here, since tokenId will always be one of the "extreme" values of the range.
how would we decrease the end or increase the start (unlimited supply) on every transfer before remaining only 1 token if we enter this condition only when is the end token (unlimited) or the start token?

Reply:
You are right, I guess I got confused because I was working on token gating and we don't decrease/increase ranges there. Thanks for the explanation.